askama/src/filters/std.rs

```diff
@@ -0,0 +1,61 @@
+use std::collections::HashSet;
+use std::hash::Hash;
+use std::rc::Rc;
+
+/// Returns an iterator with all duplicates removed.
+///
+/// The sorting order is kept, no extra allocation is performed. However, to make it possible,
+/// the data is wrapped inside `Rc`.
+///
+/// ```
+/// # use askama::{Template, filters};
+/// #[derive(Template)]
+/// #[template(ext = "html", source = "{% for elem in example|unique %}{{ elem }},{% endfor %}")]
+/// struct Example<'a> {
+///     example: Vec<&'a str>,
+/// }
+///
+/// assert_eq!(
+///     Example { example: vec!["a", "b", "a", "c"] }.to_string(),
+///     "a,b,c,"
+/// );
+/// ```
+pub fn unique<T: Hash + Eq>(it: impl IntoIterator<Item = T>) -> impl Iterator<Item = Rc<T>> {
+    let mut set = HashSet::new();
+
+    it.into_iter().filter_map(move |elem| {
+        // To prevent cloning the data, we need to use `Rc`, like that we can clone `elem` as
+        // key of the `HashSet` and return it.
+        let elem = Rc::new(elem);
+
+        if set.insert(Rc::clone(&elem)) {
+            Some(elem)
+        } else {
+            None
+        }
+    })
```

Comment:
If we increase the msrv to 1.83, then we could use `VacantEntry::insert_entry()`. This way the `Rc` only needs to be cloned if the key is actually absent from the list ↓. What do you think? Would it be worth it?

```rust
let mut set = HashMap::new();

it.into_iter().filter_map(move |elem| {
    // To prevent cloning the data, we need to use `Rc`, like that we can clone `elem` as
    // key of the `HashSet` and return it.
    if let Entry::Vacant(entry) = set.entry(Rc::new(elem)) {
        Some(Rc::clone(entry.insert_entry(()).key()))
    } else {
        None
    }
})
```

Reply:
Huuuuuum... Ok let's go!